app/src/main/java/fr/free/nrw/commons/upload/PendingUploadsFragment.kt

```diff
@@ -0,0 +1,299 @@
+            String.format(Locale.getDefault(), getString(R.string.no)),
+            {
+                ViewUtil.showShortToast(context, R.string.cancelling_upload)
+                pendingUploadsPresenter.deleteUpload(contribution, this.requireContext().applicationContext)
+                CommonsApplication.cancelledUploads.add(contribution!!.pageId)
+                resetProgressBar()
+            },
+            {}
+        )
+    }
+
+    companion object {
+        @JvmStatic
+        fun newInstance(param1: String, param2: String) =
+            PendingUploadsFragment().apply {
+                arguments = Bundle().apply {
+                    putString(ARG_PARAM1, param1)
+                    putString(ARG_PARAM2, param2)
+                }
+            }
+    }
+
+    fun restartUploads() {
+        if (contributionsList != null){
+            pendingUploadsPresenter.restartUploads(contributionsList, 0 , this.requireContext().applicationContext)
+        }
+    }
+
+    fun pauseUploads() {
+        if (contributionsList != null){
+            pendingUploadsPresenter.pauseUploads(contributionsList, 0, this.requireContext().applicationContext)
+        }
+    }
+
+    fun deleteUploads(){
+        if (contributionsList != null){
+            showAlertDialog(
+                requireActivity(),
+                String.format(
+                    Locale.getDefault(),
+                    getString(R.string.cancelling_all_the_uploads)
+                ),
+                String.format(
+                    Locale.getDefault(),
+                    getString(R.string.are_you_sure_that_you_want_cancel_all_the_uploads)
+                ),
+                String.format(Locale.getDefault(), getString(R.string.yes)),
+                String.format(Locale.getDefault(), getString(R.string.no)),
+                {
+                    ViewUtil.showShortToast(context, R.string.cancelling_upload)
+                    uploadProgressActivity.hidePendingIcons()
+                    pendingUploadsPresenter.deleteUploads(contributionsList, 0, this.requireContext().applicationContext)
+                },
+                {}
+            )
+        }
+    }
+
+    /**
+     * Restarts the upload process for a contribution
+     *
+     * @param contribution
+     */
+    fun restartUpload(contribution: Contribution) {
+        contribution.state = Contribution.STATE_QUEUED
+        pendingUploadsPresenter.saveContribution(contribution, this.requireContext().applicationContext)
+        Timber.d("Restarting for %s", contribution.toString())
+    }
+
+    /**
+     * Retry upload when it is failed
+     *
+     * @param contribution contribution to be retried
+     */
+    fun retryUpload(contribution: Contribution) {
+        if (NetworkUtils.isInternetConnectionEstablished(context)) {
+            if (contribution.state == Contribution.STATE_PAUSED
+                || contribution.state == Contribution.STATE_QUEUED_LIMITED_CONNECTION_MODE
```

Comment:
How are we currently supporting limited connection mode? Is this by using the pause button in the top bar?